### PR TITLE
TsuMeta: Add editproposal command

### PR DESCRIPTION
**Pull request by DeathlyPlays**

### CODE REVIEW

File: wavelength-plugins/tsumeta.js [hunk 1]
```diff
@@ -209,17 +243,18 @@ exports.commands = {
...
+		/tsumeta alert [message] - Sends a message to all online users from the TsuMeta Council. Only for Desokoro.
+		/tsumeta propose [what you modified], [change requested] - Proposes a change for the TsuMeta metagame. Must be in the TsuMeta Council to use this command.
+		/tsumeta edit [proposal], [description update] - Edits a proposal. Must be the submitter of the proposal, Desokoro, or xcmr.
+		/tsumeta proposals [optional proposal ID] - Checks the specified proposal ID, if not specified generates a random one from the proposals index.
+		/tsumeta deleteproposal [proposal] - Deletes the specified proposal. Reserved for Desokoro and xcmr.
+		/tsumeta viewproposals - Shows the list of proposals.
+		/tsumeta suspend [target] - Suspends a user from proposing/participating in the TsuMeta council. Only for Desokoro and xcmr.
+		/tsumeta unsuspend [target] - Unsuspends a user from proposing/participating in the TsuMeta council. Only for Desokoro and xcmr.
+		/tsumeta forums - Displays the official TsuMeta Website.
+		/tsumeta help - Displays this help command.`,
```
Comment (HoeenCoder) [inline wavelength-plugins/tsumeta.js]: Why is it all one giant string now?

File: wavelength-plugins/tsumeta.js [hunk 2]
```diff
@@ -115,6 +115,40 @@ exports.commands = {
...
+			if (user.userid === "xcmr" && Db.councilmember.has("xcmr")) {
+				proposals[toId(proposal)].desc = newDesc;
+				write();
+				this.sendReplyBox(`You have successfully modified proposal <strong>${proposals[proposalid].idea}'s</strong> description to:<br />${proposals[proposalid].desc}`);
+				if (Rooms('tsumetacommittee')) {
+					Rooms('tsumetacommittee').add(`|c|~TsuMeta Council|${user.name} has edited ${proposals[proposalid].idea}'s description.`).update();
+				}
+				return true;
+			}
+			if (proposals[proposalid].creator !== user.userid && user.userid !== "desokoro") return this.errorReply(`Only the creator "${proposals[proposalid].creator}" can modify this proposal.`);
```
Comment (HoeenCoder) [inline wavelength-plugins/tsumeta.js]: merge the xcmr check into this one.
`if (proposals[proposalid].creator !== user.userid && user.userid !== "desokoro" && (user.userid !== 'xcmr' || !Db.councilmember.has("xcmr")))`

File: wavelength-plugins/tsumeta.js (hunk 1 above)
Comment (Mystifi) [reply wavelength-plugins/tsumeta.js]: I'm pretty sure that the padding after each newline stays within the template string. If so, these should be separate.

File: wavelength-plugins/tsumeta.js [hunk 3]
```diff
@@ -115,6 +115,40 @@ exports.commands = {
...
+			if (proposal.length > 500) return this.errorReply("Please keep your changes to a maximum of 500 characters.");
+			let newDesc = parts[1];
+			if (user.userid === "xcmr" && Db.councilmember.has("xcmr")) {
+				proposals[toId(proposal)].desc = newDesc;
+				write();
+				this.sendReplyBox(`You have successfully modified proposal <strong>${proposals[proposalid].idea}'s</strong> description to:<br />${proposals[proposalid].desc}`);
+				if (Rooms('tsumetacommittee')) {
+					Rooms('tsumetacommittee').add(`|c|~TsuMeta Council|${user.name} has edited ${proposals[proposalid].idea}'s description.`).update();
+				}
+				return true;
```
Comment (Mystifi) [inline wavelength-plugins/tsumeta.js]: If you want to return early, just use `return;`.

File: wavelength-plugins/tsumeta.js [hunk 4]
```diff
@@ -115,6 +115,40 @@ exports.commands = {
...
 		},
 
+		modify: "editproposal",
+		edit: "editproposal",
+		modifyproposal: "editproposal",
+		editproposal: function (target, room, user) {
+			if (!isTsuMetaCouncil(user.userid)) return this.errorReply('You are not in the TsuMeta council, or have been suspended.');
+			if (!this.canTalk()) return this.errorReply("You cannot do this while unable to talk.");
+			let parts = target.split(',');
+			for (let u in parts) parts[u] = parts[u].trim();
```
Comment (Mystifi) [inline wavelength-plugins/tsumeta.js]: Consolidate this into `let [proposal, ...newDesc] = target.split(',').map(p => p.trim());`. I never understand why panpawn doesn't use this...anyways, it allows you to remove `let proposal = parts[0]; let newDesc = parts[1];`, and it lets proposal descriptions include commas.

File: wavelength-plugins/tsumeta.js [hunk 5]
```diff
@@ -115,6 +115,40 @@ exports.commands = {
...
+			if (!parts[0] || !parts[1]) return this.parse('/tsumetahelp');
+			let proposal = parts[0];
+			let proposalid = toId(proposal);
+			if (!proposals[proposalid]) return this.errorReply(`This proposal doesn't exist!`);
+			if (proposal.length > 500) return this.errorReply("Please keep your changes to a maximum of 500 characters.");
+			let newDesc = parts[1];
+			if (user.userid === "xcmr" && Db.councilmember.has("xcmr")) {
+				proposals[toId(proposal)].desc = newDesc;
```
Comment (Mystifi) [inline wavelength-plugins/tsumeta.js]: `toId(proposal)` here should be `proposalId`.

File: wavelength-plugins/tsumeta.js [hunk 6]
```diff
@@ -115,6 +115,40 @@ exports.commands = {
...
+				write();
+				this.sendReplyBox(`You have successfully modified proposal <strong>${proposals[proposalid].idea}'s</strong> description to:<br />${proposals[proposalid].desc}`);
+				if (Rooms('tsumetacommittee')) {
+					Rooms('tsumetacommittee').add(`|c|~TsuMeta Council|${user.name} has edited ${proposals[proposalid].idea}'s description.`).update();
+				}
+				return true;
+			}
+			if (proposals[proposalid].creator !== user.userid && user.userid !== "desokoro") return this.errorReply(`Only the creator "${proposals[proposalid].creator}" can modify this proposal.`);
+			proposals[toId(proposal)].desc = newDesc;
```
Comment (Mystifi) [inline wavelength-plugins/tsumeta.js]: This should be `proposalId` again. Also, following the `newDesc` advise from beforehand, the splat operator returns an array, so you'll need to use `newDesc.join(', ')`.

File: wavelength-plugins/tsumeta.js [hunk 7]
```diff
@@ -115,6 +115,40 @@ exports.commands = {
...
+			write();
```
Comment (Mystifi) [inline wavelength-plugins/tsumeta.js]: This is simply nitpicking, but could you rename this to `writeProposals()`? It makes it clearer to read.

File: wavelength-plugins/tsumeta.js (hunk 1 above)
Comment (DeathlyPlays) [reply wavelength-plugins/tsumeta.js]: Honestly I thought it looked cleaner to read

File: wavelength-plugins/tsumeta.js (hunk 7 above)
Comment (DeathlyPlays) [reply wavelength-plugins/tsumeta.js]: Ok

**Comment by DeathlyPlays** [comment]
@HoeenCoder @Mystifi bump

File: wavelength-plugins/tsumeta.js (hunk 1 above)
Comment (Volco) [reply wavelength-plugins/tsumeta.js]: @HoeenCoder 1 giant template string with line breaks (like how it is) displays it with spacings between lines

Comment (DeathlyPlays) [reply wavelength-plugins/tsumeta.js]: Just looks a hell of a lot easier to read code-wise.

**Comment by Volco** [comment]
@DeathlyPlays fully tested? fully works?

**Comment by DeathlyPlays** [comment]
yup

**Pull request merged**